### PR TITLE
xen-dom-mgmt: Remove pages from physmap when unmapping

### PR DESCRIPTION
**Pull request by Deedone**
When unmapping a region, pages should be removed from the self physmap as well. This is necessary to avoid leaking page references and ensure that domains can be properly destroyed.

### CODE REVIEW

**Comment by GrygiriiS** [comment]
I tried it and do not see zombies anymore: `Tested-by: Grygorii Strashko <grygorii_strashko@epam.com>`

**Review by lorc — COMMENTED** [review]
Before I continue reviewing: have you tried this with `CONFIG_XEN_REGIONS=n`?

**Comment by Deedone** [comment]
>have you tried this with CONFIG_XEN_REGIONS=n
@lorc builds and works okay for me

File: xen-dom-mgmt/src/mem-mgmt.c [hunk 1]
```diff
@@ -260,7 +260,18 @@ int xenmem_unmap_region(uint64_t nr_pages, void *mapped_addr)
 		return -EFAULT;
 	}
 #else
+	uint64_t base_pfn = xen_virt_to_gfn(mapped_addr);
+	int rc, i = 0;
+
 	xen_region_unmap(mapped_addr, nr_pages);
+	for (i = 0; i < nr_pages; i++) {
```
Comment (firscity) [inline xen-dom-mgmt/src/mem-mgmt.c:267]: It seems like `region_space_remove()` that is called in other part of ifdef is doing the same thing
Maybe it can be re-arranged to be called in both cases?

Comment (lorc) [reply xen-dom-mgmt/src/mem-mgmt.c:267]: Yes, thanks. This is what I wanted to asks. It seems that `xendom_remove_from_physmap` needs to be called in both cases but I can't see where is it called in other part of `#if` block. Maybe, some we need some refactoring here...

File: xen-dom-mgmt/src/mem-mgmt.c [hunk 2]
```diff
@@ -102,19 +102,6 @@ uint64_t xenmem_populate_physmap(int domid,
 	return i;
 }
 
-#if defined(CONFIG_XEN_REGIONS)
```
Comment (lorc) [inline xen-dom-mgmt/src/mem-mgmt.c:105]: Do you really need to move those functions around? Looks like unnecessary change.

**Review by lorc — APPROVED** [review]
Looks good apart from that small nitpick.

With the comment addressed: 

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epan.com>`

File: xen-dom-mgmt/src/mem-mgmt.c (hunk 2 above)
Comment (Deedone) [reply xen-dom-mgmt/src/mem-mgmt.c:105]: I actually moved region_space_remove out of the if .. else guards.  For some reason the diff generated in a very strange way. Maybe move it farther?

**Review by firscity — APPROVED** [review]
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

File: xen-dom-mgmt/src/mem-mgmt.c (hunk 2 above)
Comment (lorc) [reply xen-dom-mgmt/src/mem-mgmt.c:105]: Ah, I see now. No, its fine. Leave it as is.
